### PR TITLE
memory: fix data race on concurrent access of Tracker.children

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

**Comment by zz-jason** [comment]
@tiancaiamao @lamxTyler @fipped PTAL

File: util/memory/tracker.go [hunk 1]
```diff
@@ -32,18 +32,19 @@ import (
 // 1. tracker.SetLabel() / tracker.SetActionOnExceed() / tracker.AttachTo()
 // 2. tracker.Consume() / tracker.ReplaceChild() / tracker.BytesConsumed()
 //
-// NOTE:
-// 1. Only "BytesConsumed()" and "Consume()" are thread-safe.
-// 2. Adjustment of Tracker tree is not thread-safe.
+// NOTE: We only protect concurrent access to "bytesConsumed" and "children",
+// that is to say:
+// 1. Only "BytesConsumed()", "Consume()" and "AttachTo()" are thread-safe.
+// 2. Ohter operations of a Tracker tree is not thread-safe.
```
Comment (coocood) [inline util/memory/tracker.go]: Other

**Comment by coocood** [comment]
LGTM

**Comment by zz-jason** [comment]
@XuHuaiyu @zhexuany PTAL

**Pull request merged**